src/panels/lovelace/editor/config-elements/hui-heading-card-editor.ts

```diff
@@ -0,0 +1,313 @@
+import { mdiGestureTap } from "@mdi/js";
+import { css, html, LitElement, nothing } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import memoizeOne from "memoize-one";
+import {
+  any,
+  array,
+  assert,
+  assign,
+  literal,
+  object,
+  optional,
+  string,
+  union,
+} from "superstruct";
+import { fireEvent, HASSDomEvent } from "../../../../common/dom/fire_event";
+import { LocalizeFunc } from "../../../../common/translations/localize";
+import "../../../../components/ha-form/ha-form";
+import type {
+  HaFormSchema,
+  SchemaUnion,
+} from "../../../../components/ha-form/types";
+import type { HomeAssistant } from "../../../../types";
+import type {
+  HeadingCardConfig,
+  HeadingCardEntityConfig,
+} from "../../cards/types";
+import { UiAction } from "../../components/hui-action-editor";
+import type { LovelaceCardEditor } from "../../types";
+import "../hui-sub-form-editor";
+import { actionConfigStruct } from "../structs/action-struct";
+import { baseLovelaceCardConfig } from "../structs/base-card-struct";
+import { SubFormEditorData } from "../types";
+import { configElementStyle } from "./config-elements-style";
+import "./hui-entities-editor";
+
+const actions: UiAction[] = ["navigate", "url", "perform-action", "none"];
+
+const cardConfigStruct = assign(
+  baseLovelaceCardConfig,
+  object({
+    heading_style: optional(union([literal("title"), literal("subtitle")])),
+    heading: optional(string()),
+    icon: optional(string()),
+    tap_action: optional(actionConfigStruct),
+    entities: optional(array(any())),
```

Comment:
here entities is set to `any`, but below is a struct for entities? Why not use it?

Do we support both a string and object for entities? Because the editor seems to not work when using a string.

Reply:
I do not prefer to put the full editor in yaml mode if there is a problem with an entity.